pkgs/development/compilers/vala/default.nix

```diff
@@ -1,11 +1,46 @@
-{ stdenv, lib, fetchurl, pkgconfig, flex, bison, libxslt, autoconf, automake, graphviz
-, glib, libiconv, libintl, libtool, expat
+{ stdenv, lib, fetchurl, fetchpatch, pkgconfig, flex, bison, libxslt, autoconf, automake, autoreconfHook
+, graphviz, glib, libiconv, libintl, libtool, expat
 }:
 
 let
-  generic = { major, minor, sha256, extraNativeBuildInputs ? [], extraBuildInputs ? [] }:
+  generic = lib.makeOverridable ({
+    major, minor, sha256,
+    extraNativeBuildInputs ? [],
+    extraBuildInputs ? [],
+    withGraphviz ? false
+  }:
   let
     atLeast = lib.versionAtLeast "${major}.${minor}";
+
+    # Patches from the openembedded-core project to build vala without graphviz
+    # support. We need to apply an additional patch to allow building when the
+    # header file isn't available at all, but that patch (./gvc-compat.patch)
+    # can be shared between all versions of Vala so far.
+    graphvizPatch =
+      let
+        fp = { commit, sha256 }: fetchpatch {
+          url = "https://github.com/openembedded/openembedded-core/raw/${commit}/meta/recipes-devtools/vala/vala/disable-graphviz.patch";
+          inherit sha256;
+        };
+
+      in {
+        "0.38" = fp {
+          commit = "2c290f7253bba5ceb0d32e7d0b0ec0d0e81cc263";
+          sha256 = "056ybapfx18d7xw1k8k85nsjrc26qk2q2d9v9nz2zrcwbq5brhkp";
+        };
+
+        # NOTE: the openembedded-core project doesn't have a patch for 0.40.12
+        # or 0.42.4 just yet; we've fixed the single merge conflict in the
+        # patches below and checked them in here.
+        #     0.40.12: https://github.com/openembedded/openembedded-core/raw/8553c52f174af4c8c433c543f806f5ed5c1ec48c/meta/recipes-devtools/vala/vala/disable-graphviz.patch
+        #     0.42.4:  https://github.com/openembedded/openembedded-core/raw/dfbbff39cfd413510abbd60930232a9c6b35d765/meta/recipes-devtools/vala/vala/disable-graphviz.patch
+        "0.40" = ./disable-graphviz-0.40.12.patch;
+        "0.42" = ./disable-graphviz-0.42.4.patch;
```

Comment:
Any reason not to fetch these from the above URLs? Not that it's a blocker, but I believe it's preferred to fetch patches if possible.

Reply:
>NOTE: the openembedded-core project doesn't have a patch for 0.40.12
  or 0.42.4 just yet; **we've fixed the single merge conflict in the
 patches below and checked them in here**.